Userland/Libraries/LibJS/Runtime/RegExpLegacyStaticProperties.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2022, LI YUBEI <leeight@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibJS/Runtime/FunctionObject.h>
+#include <LibJS/Runtime/PrimitiveString.h>
+#include <LibJS/Runtime/RegExpConstructor.h>
+#include <LibJS/Runtime/VM.h>
+#include <LibJS/Runtime/Value.h>
+namespace JS {
+
+void RegExpLegacyStaticProperties::invalidate()
+{
+    m_input = {};
+    m_last_match = {};
+    m_last_paren = {};
+    m_left_context = {};
+    m_right_context = {};
+    m_$1 = {};
+    m_$2 = {};
+    m_$3 = {};
+    m_$4 = {};
+    m_$5 = {};
+    m_$6 = {};
+    m_$7 = {};
+    m_$8 = {};
+    m_$9 = {};
+}
+
+// GetLegacyRegExpStaticProperty( C, thisValue, internalSlotName ), https://github.com/tc39/proposal-regexp-legacy-features#getlegacyregexpstaticproperty-c-thisvalue-internalslotname-
+ThrowCompletionOr<Value> get_legacy_regexp_static_property(VM& vm, RegExpConstructor& constructor, Value this_value, Optional<String> (RegExpLegacyStaticProperties::*property_getter)() const)
+{
+    // 1. Assert C is an object that has an internal slot named internalSlotName.
+
+    // 2. If SameValue(C, thisValue) is false, throw a TypeError exception.
+    if (!same_value(&constructor, this_value))
+        return vm.throw_completion<TypeError>(ErrorType::GetLegacyRegExpStaticPropertyThisValueMismatch);
+
+    // 3. Let val be the value of the internal slot of C named internalSlotName.
+    auto val = (constructor.legacy_static_properties().*property_getter)();
+
+    // 4. If val is empty, throw a TypeError exception.
+    if (!val.has_value())
+        return vm.throw_completion<TypeError>(ErrorType::GetLegacyRegExpStaticPropertyValueEmpty);
+
+    // 5. Return val.
+    return js_string(vm, val.value());
+}
+
+// SetLegacyRegExpStaticProperty( C, thisValue, internalSlotName, val ), https://github.com/tc39/proposal-regexp-legacy-features#setlegacyregexpstaticproperty-c-thisvalue-internalslotname-val-
+ThrowCompletionOr<void> set_legacy_regexp_static_property(VM& vm, RegExpConstructor& constructor, Value this_value, void (RegExpLegacyStaticProperties::*property_setter)(String const&), Value value)
```

Comment:
Need to pass by value for the `move()` to take effect

```suggestion
ThrowCompletionOr<void> set_legacy_regexp_static_property(VM& vm, RegExpConstructor& constructor, Value this_value, void (RegExpLegacyStaticProperties::*property_setter)(String), Value value)
```